src/runtime/refiner.ts

```diff
@@ -228,6 +286,20 @@ export class BinaryExpression extends RefinementExpression {
       default: return this;
     }
   }
+
+  containsField(fieldName: string): boolean {
+    return this.leftExpr.containsField(fieldName) || this.rightExpr.containsField(fieldName);
+  }
+
+  fieldNameIfUnivariate(): string {
```

Comment:
How about we just get out the `Set` of field names here? They we can reuse this later?

Reply:
How about this?